### PR TITLE
client: avoid adding checked inode to the delayed_caps

### PR DESCRIPTION
**Pull request by taodd**
no need to check again in delayed_caps

Signed-off-by: dongdong tao <tdd21151186@gmail.com>

### CODE REVIEW

File: src/client/Client.cc [hunk 1]
```diff
@@ -3447,6 +3446,7 @@ void Client::check_caps(Inode *in, unsigned flags)
 
     if (now < in->hold_caps_until) {
       ldout(cct, 10) << "delaying cap release" << dendl;
+      delayed_caps.push_back(&in->cap_item);
```
Comment (ukernel) [inline src/client/Client.cc]: remove above code and change this code to:

if (!(flags & CHECK_CAPS_NODELAY)) {
  ldout(cct, 10) << "delaying cap release" << dendl;
  in->hold_caps_until = now
  in->hold_caps_until += cct->_conf->client_caps_release_delay;
  delayed_caps.push_back(&in->cap_item);
}

I think it's more clear

Comment (taodd) [reply src/client/Client.cc]: greate point ! 👍

**Comment by taodd** [comment]
the only thing is that, your way might set hold_cap_until multiple times.

**Comment by ukernel** [comment]
we can use a local variable to avoid that

**Comment by taodd** [comment]
@ukernel it's more clean now, please have a look again.

**Comment by taodd** [comment]
@batrick updated

**Pull request merged**